x/authority/keeper/msg_server_update_policies.go

```diff
@@ -0,0 +1,30 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/zeta-chain/zetacore/x/authority/types"
+)
+
+// UpdatePolicies updates policies
+func (k msgServer) UpdatePolicies(goCtx context.Context, msg *types.MsgUpdatePolicies) (*types.MsgUpdatePoliciesResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check called by governance
+	if k.govAddr.String() != msg.Signer {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority, expected %s, got %s",
+			k.govAddr.String(),
+			msg.Signer,
+		)
+	}
+
+	// set policies
+	k.SetPolicies(ctx, msg.Policies)
```

Comment:
To clarify, this will be a full update on existing policies, which allows gov proposals to delete an existing policy? A quick question to my mind was that do we have any MUST-HAVE policy that can't be deleted?

Reply:
This is the current architecture we have right now, any policy can be deleted. If this is the case then the related message just can't be executed.